### ISSUE TITLE
Raycasting picking up non_ray_pickable CollisionObjects

### ISSUE THREAD

**Issue by paolobb4** [godotengine/godot#7108]
**Operating system or device - Godot version:**
Archlinux 64-bit with xfce4, using godot_v2.1-stable_x11.64

**Issue description** (what happened, and what was expected):
Doing raycast from script, collided with a _CollisionObject_ that had the _'ray_pickable'_ attribute set to false. That obviously shoudn't happen.

**Steps to reproduce:**
Set an object of any type, let's call it _obj1_, a little higher in the Y-axis from a _CollitionObject_ of any type, let's call it _obj2_, wich has a _CollisionShape_. Now execute this from script:
```
var obj = get_node("obj1")
var pos = obj.get_translation()
	
var spaceState = get_world().get_direct_space_state()
# Casting from ten units down the Y-axis
var castResult = spaceState.intersect_ray(pos, pos-Vector3(0, 10, 0))
	
if not castResult.empty():
	print("Collision occured")
else:
	print("No collision occured")
```
As an extra detail, the issue persists even if you delete the _CollisionShape_ from _obj2_.

**Link to minimal example project**:
https://bitbucket.org/PaoloBB4/godot-test-raycasting-error
In this example there are two objects _block1_ and _block2_, colored red and green accordingly and _block2_ is a _StaticBody_ with _'ray_pickable'_ set to false.

If collision occures the red block is translated to the point of collision and the message _"Collition occured"_ is displayed on output.
Otherwise it remains in place and the message _"No collition occured"_ is displayed on output.

**Comment by vnen** [comment]
The C++ `intersect_ray` function has a `pick_ray` argument which is `false` by default and is not exposed to GDScript. Since it is not a "pick ray", it ignores the `ray_pickable` property.

**Comment by paolobb4** [comment]
Oh ok, that makes sense. So I gess we'll just have to wait for the next update (?
I'm working on a small project so it was easy to work around it, but it's preaty ugly having to work around such simple things.
Thanks so much though, I feel really relieved now I know why it wasn't working.

**Comment by Noshyaar** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
As there was no answer to the above request, we're closing this issue as inactive.
If you can still reproduce it, please open a new issue against a currently supported release of Godot.
